express-notes-api/server.js

```diff
@@ -0,0 +1,113 @@
+import express from 'express';
+import { readFile, writeFile } from 'node:fs/promises';
+
+const app = express();
+
+app.use(express.json());
+
+// gets a list of notes in an array
+app.get('/api/notes', async (req, res) => {
+  const data = await fetchFile();
+  const array = [];
+
+  for (const note in data.notes) {
+    array.push(data.notes[note]);
+  }
+  res.status(200).send(array);
+});
+
+// gets a single note by id
+app.get('/api/notes/:id', async (req, res) => {
+  const data = await fetchFile();
+  const id = req.params.id;
+
+  idErrorHandler(id, data, res);
```

Comment:
If this sends an error response, you will still execute the next line of code and attempt to return a 200. You need to check if `idErrorHandler` sends an error response, and then stop executing this function.